spec/system/teams_spec.rb

```diff
@@ -142,6 +143,56 @@
     end
   end
 
+  context "#destroy" do
+    let(:delete_migrate_btn) { "Migrate namespaces and delete team" }
+
+    before do
+      APP_CONFIG["delete"] = { "enabled" => true }
+      visit team_path(team)
+      toggle_team_delete_modal
+    end
+
+    it "deletes team with namespaces" do
+      click_button "Delete team and its namespaces"
+
+      expect(page).to have_content("Team '#{team.name}' was removed successfully")
+    end
+
+    it "deletes team without namespaces" do
+      visit team_path(team2)
+      toggle_team_delete_modal
+      click_button "I understand, delete team"
+
+      expect(page).to have_content("Team '#{team2.name}' was removed successfully")
+    end
+
+    it "deletes team migrating all its namespaces" do
+      select_vue_multiselect(".team_select", team2.name)
+      click_button delete_migrate_btn
+
+      expect(page).to have_content("Team '#{team.name}' was removed successfully and its "\
+        "namespaces were migrated to '#{team2.name}'")
+      expect(page).to have_current_path(teams_path)
+
+      visit team_path(team2)
+      expect(page).to have_link(namespace.name)
+    end
+
+    it "cannot migrate namespaces to the same team" do
+      select_vue_multiselect(".team_select", team.name)
+
+      expect(page).to have_content("You cannot select the original team")
+      expect(page).to have_button(delete_migrate_btn, disabled: true)
```

Comment:
I've checked the previous PR, and I see that we do not enforce this on the API level. I think we should.

Reply:
Fixed.